.github/workflows/rust.yml

```
@@ -90,28 +88,15 @@ jobs:
         TARGET: x86_64-unknown-linux-gnu
       run: sh ci/run.sh
 
-  # These jobs doesn't actually test anything, but they're only used to tell
-  # bors the build completed, as there is no practical way to detect when a
-  # workflow is successful listening to webhooks only.
-  #
-  # ALL THE PREVIOUS JOBS NEED TO BE ADDED TO THE `needs` SECTION OF THIS JOB!
-
-  end-success:
-    name: bors build finished
-    if: github.event.pusher.name == 'bors' && success()
-    runs-on: ubuntu-latest
-    needs: [miri, rustfmt_clippy, test, msrv]
-
-    steps:
-      - name: Mark the job as successful
-        run: exit 0
-
-  end-failure:
-    name: bors build finished
-    if: github.event.pusher.name == 'bors' && (failure() || cancelled())
+  conclusion:
+    needs: [test, msrv]
+    # !cancelled() executes the job regardless of whether the previous jobs passed or failed.
+    # Otherwise, this job would be skipped by default.
```

```suggestion
    # !cancelled() executes the job regardless of whether the previous jobs passed, failed or get skipped.
```
I think this is clearer, what do you think?

---

It's a bit subtle in that if a job in `needs` fails, **this job** gets skipped, which we want to avoid. So the issue is not so much the previous jobs being skipped, but them failing. But your description is indeed more general. I changed it to that.